R/combine_checkboxes.R

```diff
@@ -0,0 +1,192 @@
+#'   values_to = "race_combined",
+#'   multi_value_label = "Multiple",
+#'   values_fill = NA
+#' )
+#' }
+#'
+#' @export
+
+combine_checkboxes <- function(supertbl,
+                               tbl,
+                               cols,
+                               values_to,
+                               multi_value_label = "Multiple",
+                               values_fill = NA,
+                               raw_or_label = "label",
+                               keep = TRUE) {
+
+  # Check args ---
+  check_arg_is_supertbl(supertbl, req_cols = c("redcap_data", "redcap_metadata"))
+  check_arg_is_character(tbl, len = 1, any.missing = FALSE)
+  check_arg_is_character(values_to, len = 1, any.missing = FALSE)
+  check_arg_is_character(multi_value_label, len = 1, any.missing = TRUE)
+  check_arg_is_character(values_fill, len = 1, any.missing = TRUE)
+  check_arg_choices(raw_or_label, choices = c("label", "raw"))
+  check_arg_is_logical(keep, len = 1, any.missing = FALSE)
+
+  # Save user cols to enquosure
+  cols_exp <- enquo(cols)
+
+  # Extract tbl from supertbl
+  data_tbl <- supertbl %>%
+    extract_tibble(tbl)
+
+  # Get field names from cols_exp, check that fields exist
+  field_names <- names(eval_select(cols_exp, data = data_tbl))
+  check_fields_exist(fields = field_names, expr = cols_exp)
+
+  # Assume the first instrument in the metadata contains IDs
+  # REDCap enforces this constraints, we reflect this in read_redcap -> get_field_to_drop
+  record_id_field <- supertbl$redcap_metadata[supertbl$redcap_form_name == tbl][[1]]$field_name[1]
+
+  # Combine record identifier with remaining possible project identifiers
+  instrument_identifiers <- c(
+    record_id_field,
+    "redcap_form_instance",
+    "redcap_form_name",
+    "redcap_event",
+    "redcap_event_instance"
+  )
+
+  # Define values_to as the count of TRUEs/1s for the given checkbox field
+  # Assign TRUE if multiple selections made, and FALSE if one or zero made
+  data_tbl_mod <- data_tbl %>%
+    select(any_of(instrument_identifiers), !!!eval_select(cols_exp, data_tbl)) %>%
+    mutate(
+      !!values_to := case_when(
+        rowSums(select(., eval_tidy(cols_exp))) > 1 ~ TRUE,
+        TRUE ~ FALSE
+      )
+    )
+
+  # Get metadata reference table, check that chosen fields are checkboxes
+  metadata <- get_metadata_ref(data_tbl_mod, supertbl, tbl, instrument_identifiers)
+
+  # Replace TRUEs/1s with raw/label values from metadata
+  data_tbl_mod <- data_tbl_mod %>%
+    mutate(across(
+      -c(any_of(instrument_identifiers), !!values_to),
+      ~ replace_true(.x,
+        cur_column(),
+        metadata = metadata,
+        raw_or_label = raw_or_label
+      )
+    ))
+
+  # Convert values_to from TRUE/FALSE to multi_value_label or identified single val
+  data_tbl_mod <- data_tbl_mod %>%
+    mutate(
+      across(field_names, as.character), # enforce to character strings
+      across(!!values_to, ~ as.character(.))
+    ) %>%
+    rowwise() %>%
```

Comment:
Why does this need to be done `rowwise`?

Reply:
Removing, this was leftover from a previous implementation and doesn't affect anything now but also doesn't hurt anything so I overlooked it. Thanks for catching.